src/app/components/routemap/RouteMap.tsx

```diff
@@ -0,0 +1,380 @@
+
+const RouteTextContainer =  styled.div<{lang:string}>`
+    ${tw`whitespace-nowrap text-center tracking-tighter mt-1 hm:mt-2 hsm:mt-2 font-semibold`}
+    ${(props) => props.lang === 'ko' ? tw`text-[15px] hm:text-[13px] hsm:text-[12px]` : tw`wide:text-[13px] mwide:text-[12px] text-[11px] hm:text-[11px] hsm:text-[9.3px]`}
+`
+
+const RouteStations = styled.div`
+    ${tw`transition duration-150 ease-in-out flex relative`}
+`
+
+const RouteMethod = styled.div`
+    ${tw`text-center rounded-full py-1 w-16 text-sm hm:w-12 hm:text-xs self-center text-black tracking-tight font-semibold`}
+`
+
+const MainContainer = styled.div<{status: string}>`
+    ${tw`transition duration-150 ease-in-out mx-auto h-[14rem]`}
+    ${(props) => props.status === 'entered' || props.status === 'exit' ? tw`opacity-100` : tw`opacity-0`}
+`
+
+const RouteLine = styled.div`
+    ${tw`absolute transition duration-150 ease-in-out h-[3px] z-0`}
+`
+
+const SpecialStopsText = styled.p<{key:number, lang:string}>`
+    ${tw`absolute text-xs top-[-17px] left-[-15px] text-center w-10 font-bold`}
+    ${(props) => props.lang === 'ko' ? tw`tracking-tight` : tw`tracking-tighter text-[0.7rem]`}
+`
+
+export const RouteMap = (props: {
+    status:string,
+    tab: string
+}) => {
+    const timetable = useTimeTableContext().timetable
+
+    const { t, i18n } = useTranslation()
+    // dots
+    const [direct, setDirect] = useState<JSX.Element[]>([])
+    const [cycle, setCycle] = useState<JSX.Element[]>([])
+    const [yesulin, setYesulin] = useState<JSX.Element[]>([])
+    const [jungang, setJungang] = useState<JSX.Element[]>([])
+    // lines
+    const [dirLine, setDirLine] = useState<JSX.Element[]>([])
+    const [cycLine, setCycLine] = useState<JSX.Element[]>([])
+    const [yesLine, setYesLine] = useState<JSX.Element[]>([])
+    const [junLine, setJunLine] = useState<JSX.Element[]>([])
+    // dot refs
+    const refdir = useRef<Array<HTMLDivElement>>([])
+    const refcyc = useRef<Array<HTMLDivElement>>([])
+    const refyes = useRef<Array<HTMLDivElement>>([])
+    const refjun = useRef<Array<HTMLDivElement>>([])
+    // line refs
+    const linedir = useRef<Array<HTMLDivElement>>([])
+    const linecyc = useRef<Array<HTMLDivElement>>([])
+    const lineyes = useRef<Array<HTMLDivElement>>([])
+    const linejun = useRef<Array<HTMLDivElement>>([])
+    // langtest
+    const lang = useRef<Array<HTMLParagraphElement | null>>([])
+
+    const isPrevStop = useCallback((line: string, index: number) => {
+        return props.tab === 'shuttlecoke_o' ? index !== 0 :
+            props.tab === 'shuttlecoke_i' ? line === 'direct' ? index >= 3 : index >= 4 :
+            props.tab === 'subway' ? line === 'yesulin' ? false : index >= 2 :
+            props.tab === 'yesulin' ? line === 'direct' || line === 'jungang' ? false : index >= 3 :
+            props.tab === 'jungang' ? line === 'jungang' ? index >= 3 : false : true
+    }, [props.tab])
+
+    const directInput = useCallback(() => {
```

Comment:
The `directInput` is not depending on any of the states from the parent to render its elements. This means that this component can be independent from its parent and can be as a separate component.

If this element becomes independent from its parent(RouteMap), then it would be free from the parent's rendering. This means that there is no need to wrap this whole thing with Callback. Also, it would be free from useRef that the parent is using above. Those logic would not be necessary.

The components should be responsible to one, and only one as mentioned in [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle) in SOLID principle.